demux/demux_mkv.c

```diff
@@ -1862,6 +1869,45 @@ static int demux_mkv_open_sub(demuxer_t *demuxer, mkv_track_t *track)
     sh->codec->extradata = track->private_data;
     sh->codec->extradata_size = track->private_size;
 
+    if (!strcmp(sh->codec->codec, "arib_caption")) {
+        struct AVCodecParameters **lav = talloc_ptrtype(track, lav);
+        if (!lav)
+            return 1;
+
+        talloc_set_destructor(lav, avcodec_par_destructor);
+
+        *lav = sh->codec->lav_codecpar = avcodec_parameters_alloc();
+        if (!sh->codec->lav_codecpar)
```

Comment:
If we follow the classic mpv OOM handling style, probably `MP_HANDLE_OOM(*lav)`.

I would also recommend taking a pointer towards the codecpar to not have to have `sh->codec->lav_codecpar->` everywhere as it's poked in multiple spots?